app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -44,12 +44,12 @@ def instances
       instance["volumeIDs"].each do |vol_id|
         persister.disks.build(
           :hardware        => ps_hw,
-          :device_name     => vol_id,
-          :device_type     => "block",
+          :device_name     => collector.volume(vol_id)["name"],
```

Comment:
If we are not able to find the volume by id for whatever reason (volume was deleted after we collected the VM) this will fail trying to dereference `nil["name"]` and the entire refresh will fail.

Also `collector.volume()` doesn't cache the result so this looks like it is actually making 3 API calls.

Would recommend doing `volume = collector.volume(vol_id)` at the top then you can do `name, disk_type, size = volume&.values_at("name", "diskType", "size")`